src/codec.js

```diff
@@ -69,6 +69,42 @@ Int.prototype.valueOf = function() {
 
 codec.Int = Int;
 
+var TYPE = Symbol();
+
+codec.TYPE = TYPE;
+
+function Struct() {
+  var struct = [];
+
+  struct[TYPE] = Struct.TYPE;
+
+  Object.defineProperty(struct, 'toJSON', {
+    enumerable: false,
+    value: codec.generateToJSONFromRow(struct),
+  });
+
+  return struct;
+}
+
+Struct.TYPE = 'struct';
+
+Struct.fromJSON = function(json) {
+  var struct = new Struct();
+
+  Object.keys(json || {}).forEach(function(name) {
+    var value = json[name];
+    struct.push({name, value});
+  });
+
+  return struct;
+};
+
+Struct.isStruct = function(thing) {
+  return !!(thing && thing[TYPE] === Struct.TYPE);
```

Comment:
I think just `thing && thing[TYPE] === Struct.TYPE` will always return a bool, so the `!!()` wrapper isn't necessary?

Reply:
That's not true actually, if `thing` is `null` then it would return `null` since the first expression is falsey.